client/v3/retry_interceptor.go

```diff
@@ -157,7 +157,8 @@ func (c *Client) shouldRefreshToken(err error, callOpts *options) bool {
 		// which is possible when the client token is cleared somehow
 		return c.authTokenBundle != nil // equal to c.Username != "" && c.Password != ""
 	}
-	return callOpts.retryAuth && rpctypes.Error(err) == rpctypes.ErrInvalidAuthToken
+
+	return callOpts.retryAuth && (errors.Is(err, rpctypes.ErrInvalidAuthToken) || errors.Is(err, rpctypes.ErrGRPCAuthOldRevision))
```

Comment:
@r-ashish could you check this path? I think we should use `errors.Is()` instead of `==` here.

Reply:
also line 155 would need change too. I'll add a unit test to the file for ensuring that the error handling logic is correct

Reply:
yeah agree, looks okay.